streaming/base/util.py

```diff
@@ -88,3 +93,43 @@ def bytes_to_int(bytes_str: str) -> int:
                 f'Unsupported value/suffix {bytes_str}. Supported suffix are ',
                 f'{["b"] + list(units.keys())}.'
             ]))
+
+
+def clean_stale_shared_memory() -> None:
+    """Clean up all the leaked shared memory.
+
+    In case of a distributed run, clean up happens on local rank 0 while other local ranks wait for
+    the local rank 0 to finish.
+    """
+    # Initialize torch.distributed ourselves, if necessary.
+    destroy_dist = maybe_init_dist()
+
+    # Perform clean up on local rank 0
+    if get_local_rank() == 0:
+        shm_names = [
```

Comment:
maybe put this list in constants file?

Reply:
Done in https://github.com/mosaicml/streaming/pull/299/commits/56368cb07a7657e6e52aef95fb725404c509e6bf.